app/uk/gov/hmrc/agentservicesaccount/views/ViewOps.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agentservicesaccount.views
+
+import uk.gov.hmrc.agentmtdidentifiers.model.Arn
+
+object ViewOps {
+  implicit class ArnOps(arn: Arn) {
+    def prettify: String = {
```

Comment:
What about moving it to our library?

Reply:
Raveen and I talked about this one at the time, wondering whether we should have this prettify method and some related normalisation methods (from spaced/formatted ARN strings to Arn types) put in the library. Similarly for the formatted UTRs.

We came to the conclusion that it would be nice to have it in the library primarily for avoiding code duplication. The main concern we had was that we'd then shackled to that specific formatting provided by the library, such that any potential change in the library affects the display in all services that use it. I think this is something that's mentioned in this architectural note: https://confluence.tools.tax.service.gov.uk/display/AR/Domain+Library

Another thought was that the presentation logic felt like it should belong in, and be in control of, the frontend services themselves, leaving the library with concerns solely around the identifier's non-view logic.

But the obvious con is that there's code duplication in several services.